clients/java/signalr/src/test/java/HubConnectionTest.java

```diff
@@ -1,13 +1,92 @@
 // Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
+import com.google.gson.JsonArray;
 import org.junit.Test;
 
 import static org.junit.Assert.*;
 
 public class HubConnectionTest {
-    // Coming soon!
     @Test
-    public void testEmptyCollection() {
+    public void checkHubConnectionState() throws InterruptedException {
+        Transport mockTransport = new MockEchoTransport();
+        HubConnection hubConnection = new HubConnection("test", mockTransport);
+            hubConnection.start();
+            assertTrue(hubConnection.connected);
+
+            hubConnection.stop();
+            assertFalse(hubConnection.connected);
+    }
+
+    @Test
+    public void SendWithNoParamsTriggersOnHandler() throws InterruptedException {
+        TestObject obj = new TestObject();
```

Comment:
this should just be an int

Reply:
Same here: https://github.com/aspnet/SignalR/pull/2550#discussion_r198643231

Reply:
Ok, make a local object then? `var wrapper = new Object(){ int counter = 0; };`

Reply:
Java doesn't support anonymous types in the same way C# does. That wouldn't work.

Reply:
My concern here is that `TestObject` will become this ugly monolith with 30 different properties/methods to support all tests. It's already bad looking with 2 properties used for different tests. Is there a reason `AtomicReference<T>` wasn't used?

Reply:
I thought there  would be push back against using `AtomicReference<T>` but you make a good point. The pr updated and `TestObject` has been removed